wallet/core/src/wasm/message.rs

```diff
@@ -0,0 +1,40 @@
+use crate::message::*;
+use kaspa_consensus_wasm::{PrivateKey, PublicKey};
+use secp256k1::XOnlyPublicKey;
+use std::str::FromStr;
+use wasm_bindgen::prelude::*;
+use workflow_wasm::prelude::ref_from_abi;
+
+#[wasm_bindgen(js_name = signMessage)]
+pub fn js_sign_message(raw_msg: String, privkey: JsValue) -> String {
+    let mut privkey_bytes = [0u8; 32];
+
+    if let Ok(privkey) = ref_from_abi!(PrivateKey, &privkey) {
```

Comment:
Most primitives like Private and Public key have `TryFrom<JsValue>` to automate their conversion:
https://github.com/kaspanet/rusty-kaspa/blob/master/consensus/wasm/src/keypair.rs#L170
So intake can be reduced to `let privkey = PrivateKey::try_from(privkey)?;

Reply:
Simplified. I can't use `?` here because the return values aren't `Result` or `Option` so I use `unwrap_throw()` instead.

Reply:
The result of the function should be `Result<String>`, then you can use `?`.  This is the common practice (wasm_bindgen will resolve ordinary types vs Results and setup appropriate handling).

Reply:
Fixed